test/features/OrbitalFeature_tests.jl

```diff
@@ -0,0 +1,31 @@
+using Test
+using DataFrames
+using ..ChemistryFeaturization.Atoms: elements
+using ..ChemistryFeaturization.FeatureDescriptor: OrbitalFeatureDescriptor
+using SparseArrays
+
+@testset "Encode-Decode" begin
+    ofd = OrbitalFeatureDescriptor()
+
+    @testset "Encode - with `default_ofd_encode`" begin
+        silica = AtomGraph(Float32.([0 1 0; 1 0 1; 0 1 0]), ["O", "Si", "Ar"])
+        encoded_silica = encode(ofd, silica)
+        I, J, V = findnz(encoded_silica)
+
+        @test I == [2, 3, 4, 5, 4, 5]
+        @test J == [1, 1, 2, 2, 3, 3]
+        @test V == [2, 4, 2, 2, 2, 6]
+
+        # non existent element - doesn't really make sense
+        nonexistent = AtomGraph(Float32.([1 1; 1 1]), ["Non", "Existent"])
+        @test_throws AssertionError ofd(nonexistent)
+    end
+
+    @testset "Decode - with `default_ofd_decode`" begin
```

Comment:
this one works when I run it, but when I try on a different made-up thing (my same triangle graph from the other tests but with different elements, I get an error:
```
julia> ag = AtomGraph(adjmat, ["Ne", "Fe", "Si"])
AtomGraph  with 3 nodes, 3 edges
	atoms: ["Ne", "Fe", "Si"]

julia> decode(ofd, ofd(ag))
ERROR: BoundsError: attempt to access 0-element view(::Vector{String}, Int64[]) with eltype String at index [1]
Stacktrace:
 [1] throw_boundserror(A::SubArray{String, 1, Vector{String}, Tuple{Vector{Int64}}, false}, I::Tuple{Int64})
   @ Base ./abstractarray.jl:651
 [2] checkbounds
   @ ./abstractarray.jl:616 [inlined]
 [3] getindex(V::SubArray{String, 1, Vector{String}, Tuple{Vector{Int64}}, false}, I::Int64)
   @ Base ./subarray.jl:275
 [4] default_ofd_decode(encoded_features::SparseArrays.SparseMatrixCSC{Int16, Int16})
   @ ChemistryFeaturization.FeatureDescriptor ~/git_repos/ChemistryFeaturization.jl/src/features/orbitalfeature.jl:65
 [5] (::ChemistryFeaturization.Codec.SimpleCodec)(ofd::OrbitalFeatureDescriptor, encoded_features::SparseArrays.SparseMatrixCSC{Int16, Int16})
   @ ChemistryFeaturization.FeatureDescriptor ~/git_repos/ChemistryFeaturization.jl/src/features/orbitalfeature.jl:27
 [6] decode(fd::OrbitalFeatureDescriptor, encoded_feature::SparseArrays.SparseMatrixCSC{Int16, Int16})
   @ ChemistryFeaturization.FeatureDescriptor ~/git_repos/ChemistryFeaturization.jl/src/features/features.jl:25
 [7] top-level scope
   @ REPL[33]:1
 [8] top-level scope
   @ ~/.julia/packages/CUDA/k52QH/src/initialization.jl:81
```

I don't _totally_ understand what that function is doing (some more comments might be good ;) ) but as far as I can isolate, the `shell_conf` for j=1, i=2 ends up as `"2s2.2p6.2s2.2p6.2s2.2p6.2s2.2p6.2s2.2p6.2p6."`, which leads the result of the filter on the next line to be a 0x32 SubDataFrame, so that's gotta be where that bounds error is coming from one way or another...

Reply:
Okay, so turns out, instead of having the electronic configuration for an element represented in the way it _actually_ is (like 4s orbitals are filled before 3d), pymatgen represents it in some lexical order? So, technically, the representation encoded by the encoding function was right, as it was `4s2.3d6` for `Fe`, but the lookup failed since it was stored as `3d6.4s2` in the `lookup_table` lol.
This should be relatively easy to fix up (just gotta make a couple of tweaks in `_indexorbital`)

Reply:
And yes, will add more comments too (took me a second to realize what I'd written myself haha)

Reply:
Fixed in fd42c7e.